pkg/gateway/services/container.go

```diff
@@ -13,26 +13,58 @@ import (
 func (gws GatewayService) ListContainers(ctx context.Context, in *pb.ListContainersRequest) (*pb.ListContainersResponse, error) {
 	authInfo, _ := auth.AuthInfoFromContext(ctx)
 
-	workspaceId := authInfo.Workspace.ExternalId
-
-	containerStates, err := gws.containerRepo.GetActiveContainersByWorkspaceId(workspaceId)
-	if err != nil {
-		return &pb.ListContainersResponse{
-			Ok:       false,
-			ErrorMsg: "Unable to list containers",
-		}, nil
+	type containerDetails struct {
+		WorkerId  string
+		MachineId string
 	}
 
-	containers := make([]*pb.Container, 0, len(containerStates))
+	var (
```

Comment:
Could we move this admin specific stuff out to another helper function or something?

Reply:
We can. Just updated see if it makes better sense.

Reply:
Yeah I think this is a little cleaner